Validation/RecoParticleFlow/plugins/PFCaloGPUComparisonTask.cc

```diff
@@ -0,0 +1,179 @@
+#include "DataFormats/ParticleFlowReco/interface/PFLayer.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHit.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHitFraction.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/PluginManager/interface/ModuleDef.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "SimDataFormats/GeneratorProducts/interface/HepMCProduct.h"
+
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#ifdef PFLOW_DEBUG
+#define LOGVERB(x) edm::LogVerbatim(x)
+#define LOGWARN(x) edm::LogWarning(x)
+#define LOGERR(x) edm::LogError(x)
+#define LOGDRESSED(x) edm::LogInfo(x)
+#else
+#define LOGVERB(x) LogTrace(x)
+#define LOGWARN(x) edm::LogWarning(x)
+#define LOGERR(x) edm::LogError(x)
+#define LOGDRESSED(x) LogDebug(x)
+#endif
+
+class PFCaloGPUComparisonTask : public DQMEDAnalyzer {
+public:
+  PFCaloGPUComparisonTask(edm::ParameterSet const& conf);
+  ~PFCaloGPUComparisonTask() override;
+  void analyze(edm::Event const& e, edm::EventSetup const& c) override;
+  void bookHistograms(DQMStore::IBooker&, edm::Run const&, edm::EventSetup const&) override;
+  //static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  edm::EDGetTokenT<reco::PFClusterCollection> pfClusterTok_ref_;
+  edm::EDGetTokenT<reco::PFClusterCollection> pfClusterTok_target_;
+
+  MonitorElement* pfCluster_Multiplicity_GPUvsCPU_;
+  MonitorElement* pfCluster_Energy_GPUvsCPU_;
+  MonitorElement* pfCluster_RecHitMultiplicity_GPUvsCPU_;
+  MonitorElement* pfCluster_Layer_GPUvsCPU_;
+  MonitorElement* pfCluster_Depth_GPUvsCPU_;
+  MonitorElement* pfCluster_Eta_GPUvsCPU_;
+  MonitorElement* pfCluster_Phi_GPUvsCPU_;
+
+  std::string pfCaloGPUCompDir;
+};
+
+PFCaloGPUComparisonTask::PFCaloGPUComparisonTask(const edm::ParameterSet& conf) {
+  pfClusterTok_ref_ =
+      consumes<reco::PFClusterCollection>(conf.getUntrackedParameter<edm::InputTag>("pfClusterToken_ref"));
+  pfClusterTok_target_ =
+      consumes<reco::PFClusterCollection>(conf.getUntrackedParameter<edm::InputTag>("pfClusterToken_target"));
+  pfCaloGPUCompDir = conf.getUntrackedParameter<std::string>("pfCaloGPUCompDir");
+}
+
+PFCaloGPUComparisonTask::~PFCaloGPUComparisonTask() {}
+
+void PFCaloGPUComparisonTask::bookHistograms(DQMStore::IBooker& ibooker,
+                                             edm::Run const& irun,
+                                             edm::EventSetup const& isetup) {
+  constexpr auto size = 100;
+  char histo[size];
+  ibooker.setCurrentFolder("ParticleFlow/" + pfCaloGPUCompDir);
+
+  strncpy(histo, "pfCluster_Multiplicity_GPUvsCPU_", size);
```

Comment:
How about
```cpp
std::string_view histo;
// ...
histo = "pfCluster_Multiplicity_GPUvsCPU_";
```
?

Reply:
or
```c++
const char* histo;
// ...
histo = "pfCluster_Multiplicity_GPUvsCPU_";
```
?
The histograms will make copies anyway, there is no need for the extra `strncpy`.